src/PhpSpreadsheet/Cell.php

```diff
@@ -410,7 +410,21 @@ public function setDataValidation(Cell\DataValidation $pDataValidation = null)
     }
 
     /**
-     *    Does this cell contain a Hyperlink?
+     * Does this cell contain valid value?
+     *
+     * @return bool
+     */
+    public function hasValidValue()
```

Comment:
I think we can drop entirely this method. Keeping the cell class unchanged. This limit the public API to the strict minimum for easier maintenance in the future.

Reply:
But this is part of cell as object. Isn't it?
Where is border? Which method has to be in the Cell, and which has to work as helper?

Reply:
@PowerKiKi 
So? Do you think? Or do you insist? :)
What is in priority? Maintenance or Usability?

Reply:
I had another look and the `Cell` class is not as huge as I though it was, so let's keep the method here as you suggest. But the `if (!$this->hasDataValidation())` should still be moved in the validator, otherwise the validation logic would be split into two different places.

Reply:
moved. But now we have some kind of conflict. GitHub do not allow me even view where is the problem.
Can you fix it?

Reply:
may be rebase was not so beauty, but done. :)